project-koku/koku
# 1322 remove meter

#1322 

Add foreign key to AzureMeter and AzureCostEntryProductService. Meters and product services associated with provider are deleted from database upon provider deletion.

**Testing:**
1. Add an Azure-local provider with nise data
2. Ingest/process data. Verify AzureMeter and AzureCostEntryProductService table are populated
3. Remove Azure-local provider
4. See meter and product services data removed.
5. Re-generate Nise data with the same nise yaml (hard-coded meter-id, let meter-name get randomly generated)
6. Add Azure-local provider back and ingest successfully.

**Results:**
2.  
```
postgres=# SELECT count(*) FROM acct10001.reporting_azuremeter;
-[ RECORD 1 ]
count | 4
postgres=# SELECT count(*) FROM acct10001.reporting_azurecostentryproductservice;
-[ RECORD 1 ]
count | 54
```
4. 
```
postgres=# SELECT count(*) FROM acct10001.reporting_azuremeter;
-[ RECORD 1 ]
count | 0
postgres=# SELECT count(*) FROM acct10001.reporting_azurecostentryproductservice;
-[ RECORD 1 ]
count | 0
```
6. 
```
postgres=# SELECT count(*) FROM acct10001.reporting_azuremeter;
-[ RECORD 1 ]
count | 4

postgres=# SELECT count(*) FROM acct10001.reporting_azurecostentryproductservice;
-[ RECORD 1 ]
count | 51
```
